ripple init: added context to error to show found ripple.config path.

KevM opened the pull request:
Trying to add context to an error so users can find the ripple.config that is controlling their destiny.
## Cavet

I updated the test I broke with this change but on my system I had 2 tests fail in teardown related to file/directory cleanup. Did a  `git clean -dfx` and then only one test failed. rake-d again and no tests failed. The failure is in DeleteDirectory which is a FubuCore thing. Is this a known problem?

```
Tests run: 326, Errors: 2, Failures: 0, Inconclusive: 0, Time: 8.636494 seconds
  Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0

Errors and Failures:
1) TearDown Error : ripple.Testing.from_a_solution_dir.finds_the_code_dir
   TearDown : System.IO.IOException : The directory is not empty.

--TearDown
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at FubuCore.FileSystem.DeleteDirectory(String directory) in c:\BuildAgent\work\4dafc5966c0aefb4\src\FubuCore\FileSystem.cs:line 145
   at ripple.Testing.from_a_solution_dir.TearDown() in c:\projects\ripple\src\ripple.Testing\RippleFileSystemTester.cs:line 37

2) TearDown Error : ripple.Testing.Model.SolutionFilesTester.reads_the_projects
   TearDown : System.IO.IOException : The directory is not empty.

--TearDown
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at FubuCore.FileSystem.DeleteDirectory(String directory) in c:\BuildAgent\work\4dafc5966c0aefb4\src\FubuCore\FileSystem.cs:line 145
   at ripple.Testing.Model.SolutionFilesTester.TearDown() in c:\projects\ripple\src\ripple.Testing\Model\SolutionFilesTester.cs:line 45
```

KevM committed:
ripple init: added context to error to show found ripple.config path.

Trying to add context to an error so users can find the ripple.config that is controlling their destiny.

jmarnold commented:
Thanks, Kevin. I think Jeremy's seen this as well (though I haven't). I
have an idea though.

On Thu, May 23, 2013 at 10:08 AM, Kevin Miller notifications@github.comwrote:

> Trying to add context to an error so users can find the ripple.config that
> is controlling their destiny.
> Cavet
> 
> I updated the test I broke with this change but on my system I had 2 tests
> fail in teardown related to file/directory cleanup. Did a git clean -dfxand then only one test failed. rake-d again and no tests failed. The
> failure is in DeleteDirectory which is a FubuCore thing. Is this a known
> problem?
> 
> Tests run: 326, Errors: 2, Failures: 0, Inconclusive: 0, Time: 8.636494 seconds
>   Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0
> 
> Errors and Failures:
> 1) TearDown Error : ripple.Testing.from_a_solution_dir.finds_the_code_dir
>    TearDown : System.IO.IOException : The directory is not empty.
> 
> --TearDown
>    at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
>    at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
>    at FubuCore.FileSystem.DeleteDirectory(String directory) in c:\BuildAgent\work\4dafc5966c0aefb4\src\FubuCore\FileSystem.cs:line 145
>    at ripple.Testing.from_a_solution_dir.TearDown() in c:\projects\ripple\src\ripple.Testing\RippleFileSystemTester.cs:line 37
> 
> 2) TearDown Error : ripple.Testing.Model.SolutionFilesTester.reads_the_projects
>    TearDown : System.IO.IOException : The directory is not empty.
> 
> --TearDown
>    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
>    at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
>    at FubuCore.FileSystem.DeleteDirectory(String directory) in c:\BuildAgent\work\4dafc5966c0aefb4\src\FubuCore\FileSystem.cs:line 145
>    at ripple.Testing.Model.SolutionFilesTester.TearDown() in c:\projects\ripple\src\ripple.Testing\Model\SolutionFilesTester.cs:line 45
> 
> ---
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/KevM/ripple rippleinitwithconfig
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/DarthFubuMVC/ripple/pull/88
> Commit Summary
> - ripple init: added context to error to show found ripple.config path.
> 
> File Changes
> - _M_
>   src/ripple.Testing/Commands/initializing_an_existing_ripple_solution.cshttps://github.com/DarthFubuMVC/ripple/pull/88/files#diff-0(9)
> - _M_ src/ripple/Commands/InitCommand.cshttps://github.com/DarthFubuMVC/ripple/pull/88/files#diff-1(8)
> - _M_ src/ripple/RippleFileSystem.cshttps://github.com/DarthFubuMVC/ripple/pull/88/files#diff-2(2)
> 
> Patch Links:
> - https://github.com/DarthFubuMVC/ripple/pull/88.patch
> - https://github.com/DarthFubuMVC/ripple/pull/88.diff

KevM committed:
Fixed Init command throwing when a ripple.config was not found

FindSolutionDirectory now relays optional shouldThrow argument

KevM committed:
Update filesystem tests to work outside the ripple project root

On some systems having visual studio open would cause random rake failures.

KevM committed:
Added exception to wrap CSProj xml read related exceptions

This will hopefull add more context to exceptions like this:

```
ERROR: System.Xml.XmlException: Data at the root level is invalid. Line 1, position 1.
   at System.Xml.XmlTextReaderImpl.Throw(String res, String arg)
   at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlReader.MoveToContent()
   at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XElement.Load(String uri, LoadOptions options)
   at ripple.MSBuild.CsProjFile..ctor(String filename, Solution solution) in c:\projects\ripple\src\
ripple\MSBuild\CsProjFile.cs:line 43
   at ripple.Model.Project.<>c__DisplayClass2.<.ctor>b__0() in c:\projects\ripple\src\ripple\Model\P
roject.cs:line 22
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at ripple.Model.Project.get_CsProj() in c:\projects\ripple\src\ripple\Model\Project.cs:line 28
   at ripple.Model.SolutionBuilder.<>c__DisplayClass3.<Build>b__1(Project project) in c:\projects\ri
pple\src\ripple\Model\ISolutionBuilder.cs:line 43
   at System.Collections.Generic.GenericEnumerableExtensions.Each[T](IEnumerable`1 values, Action`1
eachAction) in c:\BuildAgent\work\4dafc5966c0aefb4\src\FubuCore\EnumerableExtensions.cs:line 78
   at ripple.Model.Solution.EachProject(Action`1 action) in c:\projects\ripple\src\ripple\Model\Solu
tion.cs:line 358
   at ripple.Model.SolutionBuilder.Build() in c:\projects\ripple\src\ripple\Model\ISolutionBuilder.c
s:line 36
```

KevM commented:
The last 2 commits are an attempt to get Visual Studio not to lock up the files for most of the tests. There are still some integration tests that run under the project directory. I can't figure out how to get those moved.
